src/bblocks/datacommons_tools/custom_data/data_management.py

```diff
@@ -869,26 +898,85 @@ def validate_config(self) -> CustomDataManager:
         self._config.validate_config()
         return self
 
-    def __repr__(self) -> str:
-        input_files_count = len(self._config.inputFiles)
-        sources_count = len(self._config.sources)
-        nodes_count = sum(
-            [len(var) for var in [n.nodes for n in self._mcf_nodes.values()] if var]
-        )
+    def merge_config(
+        self,
+        config: Config | dict | str | PathLike[str],
+        *,
+        policy: DuplicatePolicy = "error",
+    ) -> CustomDataManager:
+        """Merge ``config`` into the current configuration.
 
-        variables_count = len(self._config.variables or {}) + nodes_count
-        dataframes_count = len(self._data)
+        Args:
+            config: The config to merge. This can be a Config object, a dictionary,
+                or a path to a JSON file.
+            policy: How to resolve collisions. Can be "error", "override", or "ignore".
 
-        include_input_subdirs = self._config.includeInputSubdirs
-        group_statvars = self._config.groupStatVarsByProperty
+        """
 
-        return (
-            f"<CustomDataManager config: "
-            f"\n{input_files_count} inputFiles, with {dataframes_count} containing data"
-            f"\n{sources_count} sources"
-            f"\n{variables_count} variables"
-            f"\nflags: includeInputSubdirs={include_input_subdirs}, "
-            f"groupStatVarsByProperty={group_statvars}>"
+        if isinstance(config, (str, PathLike)):
+            cfg = Config.from_json(str(config))
+        elif isinstance(config, dict):
+            cfg = Config.model_validate(config)
+        else:
+            cfg = config
+
+        merge_configs(existing=self._config, new=cfg, policy=policy)
+        return self
+
+    def merge_configs_from_directory(
```

Comment:
Is this actually needed? At least in our case I don't think we would have multiple configs in the same directory. And we can rather simply loop through `merge_config`

Reply:
Is this method actually needed?

Reply:
which one? multiple lines are highlighteed

Reply:
Is `merge_configs_from_directory` where the method iteratively merges all the configs in a directory, needed? Considering it is very unlikely that any user would have multiple configs stored in the same directory. In our case if we have multiple configs they would rather be in different directories. Users can just loop through the `merge_config` method specifying the path

Reply:
You may be reading it from the 'output' folder. Which means all the sources folders would be in there. Directory means a folder including all its subfolders.